src/handlers/colonies/colonyAdded.ts

```diff
@@ -82,8 +89,39 @@ export default async (event: ContractEvent): Promise<void> => {
     },
   });
 
+  /*
+   * Determine if the colony metadata was created from the client side
+   * by trying to fetch/refetch it up to a number of 3 blocks
+   */
+  const colonyMetadata = await tryFetchGraphqlQuery(GetColonyMetadataDocument, {
+    id: `etherealcolonymetadata-${transactionHash}`,
+  });
+
+  if (colonyMetadata?.getColonyMetadata) {
+    /*
+     * Create the colony entry in the database
+     *
+     * @TODO Move the logic from create unique colony in here, that way we can
+     * more granularly control the creation of the colony
+     *
+     * We can also do cool stuff like if there's an error creating the colony,
+     * tell the ingestor to stop watching it
+     */
+    await mutate<
+      CreateUniqueColonyMutation,
+      CreateUniqueColonyMutationVariables
+    >(CreateUniqueColonyDocument, {
+      input: {
+        colonyAddress: utils.getAddress(colonyAddress),
+        tokenAddress: utils.getAddress(tokenAddress),
+        transactionHash,
+        initiatorAddress: utils.getAddress(colonyFounderAddress),
+      },
+    });
+  }
+
```

Comment:
I thought you mentioned deleting the temporary metadata from the DB, does that happen elsewhere or not at all yet?

Reply:
That's being done in the actual `createUniqueColony` lambda: https://github.com/JoinColony/colonyCDapp/blob/refactor/colony-creation/amplify/backend/function/createUniqueColony/src/index.js#L298-L306

I agree with you now, that all the content of that lambda should be eventually moved to the block ingestor _(and the lambda removed)_. But that's a refactor for another time.